I have finished my implementation of `gbt()`, and I restored the `bilinear()`, `backward_diff()` methods with `forward_diff()` method.
The core code of `gbt()` looks like this:
```Python
num = Add(*[ T**(N-i) * c * (z-1)**i * (alpha * z + 1 - alpha)**(N-i) for c, i in zip(np[::-1], range(len(np))) ])
den = Add(*[ T**(N-i) * c * (z-1)**i * (alpha * z + 1 - alpha)**(N-i) for c, i in zip(dp[::-1], range(len(dp))) ])
```
And the code of `bilinear()` becomes this:
```Python
def bilinear(tf, sample_per):
    return gbt(tf, sample_per, 0.5)
```
And I tested the `bilinear()` method with the example in the docstring:
 ```cmd
>>> from sympy.physics.control.lti import TransferFunction, bilinear
>>> from sympy.abc import s, L, R, T
>>> tf = TransferFunction(1, s*L + R, s)
>>> numZ, denZ = gbt(tf, T, 0.5)
```
Instead of returning the correct answer,
```cmd
>>> numZ
[T, T]
>>> denZ
[2*L + R*T, -2*L + R*T]
```
The method returned me two lists with alpha multiplied, which triggered the error message in test:
```cmd
>>> numZ
[0.5*T, 0.5*T]
>>> denZ
[1.0*L + 0.5*R*T, -1.0*L + 0.5*R*T]
```

I tried to use `simplify()` to solve this problem:
```Python
from sympy import symbols, simplify
T = symbols('T')
lst = [1.0*T, 1.0*T]
simplified_lst = [simplify(expr).as_coefficients_dict()[T] for expr in last]
```
This returned me:
``` cmd
>>> simplified_lst
[1.00000000000000, 1.00000000000000]
```
And I also tried `Rational()`
```Python
from sympy import symbols, Rational
T = symbols('T')
lst = [1.0*T, 1.0*T]
rational_lst = [Rational(str(expr)) for expr in lst]
simplified_lst = [coeff.as_integer_ratio()[0] for coeff in rational_lst]
```
This returned me:
```cmd
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 1, in <listcomp>
  File "/Users/colourfulwhite/Codes/sympy/sympy/core/cache.py", line 72, in wrapper
    retval = cfunc(*args, **kwargs)
  File "/Users/colourfulwhite/Codes/sympy/sympy/core/numbers.py", line 1623, in __new__
    raise TypeError('invalid input: %s' % p)
TypeError: invalid input: 1.0*T
```
Is there any way to fix this problem?

--- a/sympy/physics/control/__init__.py
+++ b/sympy/physics/control/__init__.py
@@ -1,13 +1,13 @@
 from .lti import (TransferFunction, Series, MIMOSeries, Parallel, MIMOParallel,
-    Feedback, MIMOFeedback, TransferFunctionMatrix, sample)
+    Feedback, MIMOFeedback, TransferFunctionMatrix, gbt, bilinear, forward_diff, backward_diff)
 from .control_plots import (pole_zero_numerical_data, pole_zero_plot, step_response_numerical_data,
     step_response_plot, impulse_response_numerical_data, impulse_response_plot, ramp_response_numerical_data,
     ramp_response_plot, bode_magnitude_numerical_data, bode_phase_numerical_data, bode_magnitude_plot,
     bode_phase_plot, bode_plot)
 
 __all__ = ['TransferFunction', 'Series', 'MIMOSeries', 'Parallel',
-    'MIMOParallel', 'Feedback', 'MIMOFeedback', 'TransferFunctionMatrix','sample',
-    'backward_diff', 'pole_zero_numerical_data',
+    'MIMOParallel', 'Feedback', 'MIMOFeedback', 'TransferFunctionMatrix','gbt',
+    'bilinear', 'forward_diff', 'backward_diff', 'pole_zero_numerical_data',
     'pole_zero_plot', 'step_response_numerical_data', 'step_response_plot',
     'impulse_response_numerical_data', 'impulse_response_plot',
     'ramp_response_numerical_data', 'ramp_response_plot',

--- a/sympy/physics/control/lti.py
+++ b/sympy/physics/control/lti.py
@@ -45,7 +45,7 @@ def gbt(tf, sample_per, alpha):
     as [a, b], [c, d].
     Examples
     ========
-    >>> from sympy.physics.control.lti import TransferFunction, bilinear
+    >>> from sympy.physics.control.lti import TransferFunction, gbt
     >>> from sympy.abc import s, L, R, T
     >>> tf = TransferFunction(1, s*L + R, s)
     >>> numZ, denZ = gbt(tf, T, 0.5)
@@ -120,7 +120,7 @@ def forward_diff(tf, sample_per):
     as [a, b], [c, d].
     Examples
     ========
-    >>> from sympy.physics.control.lti import TransferFunction, bilinear
+    >>> from sympy.physics.control.lti import TransferFunction, forward_diff
     >>> from sympy.abc import s, L, R, T
     >>> tf = TransferFunction(1, s*L + R, s)
     >>> numZ, denZ = forward_diff(tf, T)

--- a/sympy/physics/control/lti.py
+++ b/sympy/physics/control/lti.py
@@ -77,10 +77,9 @@ def gbt(tf, sample_per, alpha):
     # The next line results from multiplying H(z) with z^N/z^N
 
     N = max(len(np), len(dp)) - 1
-    num = Add(*[ 1 / alpha * T**(N-i) * c * (z-1)**i * (alpha * z + 1 - alpha)**(N-i) for c, i in zip(np[::-1], range(len(np))) ])
-    den = Add(*[ 1 / alpha * T**(N-i) * c * (z-1)**i * (alpha * z + 1 - alpha)**(N-i) for c, i in zip(dp[::-1], range(len(dp))) ])
-    print(type(num))
-    print(num)
+    num = Add(*[ T**(N-i) * c * (z-1)**i * (alpha * z + 1 - alpha)**(N-i) for c, i in zip(np[::-1], range(len(np))) ])
+    den = Add(*[ T**(N-i) * c * (z-1)**i * (alpha * z + 1 - alpha)**(N-i) for c, i in zip(dp[::-1], range(len(dp))) ])
+
     num_coefs = num.as_poly(z).all_coeffs()
     den_coefs = den.as_poly(z).all_coeffs()
 

--- a/sympy/physics/control/lti.py
+++ b/sympy/physics/control/lti.py
@@ -22,7 +22,7 @@ from sympy.series import limit
 from mpmath.libmp.libmpf import prec_to_dps
 
 __all__ = ['TransferFunction', 'Series', 'MIMOSeries', 'Parallel', 'MIMOParallel',
-    'Feedback', 'MIMOFeedback', 'TransferFunctionMatrix', 'sample']
+    'Feedback', 'MIMOFeedback', 'TransferFunctionMatrix', 'gbt', 'bilinear', 'forward_diff', 'backward_diff']
 
 
 def _roots(poly, var):
@@ -33,67 +33,127 @@ def _roots(poly, var):
         r = [rootof(poly, var, k) for k in range(n)]
     return r
 
-def sample(tf, sample_per, method = 'bilinear'):
-        """
-        Returns falling coefficients of H(z) from numerator and denominator.
-        Where H(z) is the corresponding discretized transfer function,
-        discretized with the different transform method.
-        H(z) is obtained from the continuous transfer function H(s)
-        by substituting s(z) into H(s) according to different methods, where T is the
-        sample period.
-        Coefficients are falling, i.e. H(z) = (az+b)/(cz+d) is returned
-        as [a, b], [c, d].
+def gbt(tf, sample_per, alpha):
+    r"""
+    Returns falling coefficients of H(z) from numerator and denominator.
+    Where H(z) is the corresponding discretized transfer function,
+    discretized with the generalised bilinear transformation method.
+    H(z) is obtained from the continuous transfer function H(s)
+    by substituting s(z) = (z-1) / (T * (alpha*z + (1-alpha))) into H(s), where T is the
+    sample period.
+    Coefficients are falling, i.e. H(z) = (az+b)/(cz+d) is returned
+    as [a, b], [c, d].
+    Examples
+    ========
+    >>> from sympy.physics.control.lti import TransferFunction, bilinear
+    >>> from sympy.abc import s, L, R, T
+    >>> tf = TransferFunction(1, s*L + R, s)
+    >>> numZ, denZ = gbt(tf, T, 0.5)
+    >>> numZ
+    [T, T]
+    >>> denZ
+    [2*L + R*T, -2*L + R*T]
+    >>> numZ, denZ = gbt(tf, T, 0)
+    >>> numZ
+    [T, T]
+    >>> denZ
+    [2*L + R*T, -2*L + R*T]
+    >>> numZ, denZ = gbt(tf, T, 1)
+    >>> numZ
+    [T, 0]
+    >>> denZ
+    [L + R*T, -L]
+    """
+    if not tf.is_SISO:
+        raise NotImplementedError("Not implemented for MIMO systems.")
 
-        Examples
-        ========
+    T = sample_per  # and sample period T
+    s = tf.var
+    z =  s         # dummy discrete variable z
 
-        >>> from sympy.physics.control.lti import TransferFunction, sample
-        >>> from sympy.abc import s, L, R, T
-        >>> tf = TransferFunction(1, s*L + R, s)
-        >>> numZ, denZ = sample(tf, T, 'bilinear')
-        >>> numZ
-        [T, T]
-        >>> denZ
-        [2*L + R*T, -2*L + R*T]
-        >>> numZ, denZ = sample(tf, T, 'backward_diff')
-        >>> numZ
-        [T, 0]
-        >>> denZ
-        [L + R*T, -L]
-        >>> numZ, denZ = sample(tf, T, 'forward_diff')
-        >>> numZ
-        [T]
-        >>> denZ
-        [L, -L + R*T]
-        """
+    np = tf.num.as_poly(s).all_coeffs()
+    dp = tf.den.as_poly(s).all_coeffs()
 
-        if not tf.is_SISO:
-            raise NotImplementedError('Not implemented for MIMO systems.')
+    # The next line results from multiplying H(z) with z^N/z^N
 
-        T = sample_per  # and sample period T
-        s = tf.var
-        z =  s # dummy discrete variable z
+    N = max(len(np), len(dp)) - 1
+    num = Add(*[ 1 / alpha * T**(N-i) * c * (z-1)**i * (alpha * z + 1 - alpha)**(N-i) for c, i in zip(np[::-1], range(len(np))) ])
+    den = Add(*[ 1 / alpha * T**(N-i) * c * (z-1)**i * (alpha * z + 1 - alpha)**(N-i) for c, i in zip(dp[::-1], range(len(dp))) ])
+    print(type(num))
+    print(num)
+    num_coefs = num.as_poly(z).all_coeffs()
+    den_coefs = den.as_poly(z).all_coeffs()
 
-        np = tf.num.as_poly(s).all_coeffs()
-        dp = tf.den.as_poly(s).all_coeffs()
+    return num_coefs, den_coefs
 
-        # The next line results from multiplying H(z) with (z-1)^N/(z+1)^N
-        N = max(len(np), len(dp)) - 1
+def bilinear(tf, sample_per):
+    r"""
+    Returns falling coefficients of H(z) from numerator and denominator.
+    Where H(z) is the corresponding discretized transfer function,
+    discretized with the bilinear transform method.
+    H(z) is obtained from the continuous transfer function H(s)
+    by substituting s(z) = 2/T * (z-1)/(z+1) into H(s), where T is the
+    sample period.
+    Coefficients are falling, i.e. H(z) = (az+b)/(cz+d) is returned
+    as [a, b], [c, d].
+    Examples
+    ========
+    >>> from sympy.physics.control.lti import TransferFunction, bilinear
+    >>> from sympy.abc import s, L, R, T
+    >>> tf = TransferFunction(1, s*L + R, s)
+    >>> numZ, denZ = bilinear(tf, T)
+    >>> numZ
+    [T, T]
+    >>> denZ
+    [2*L + R*T, -2*L + R*T]
+    """
+    return gbt(tf, sample_per, 0.5)
 
-        if method == 'bilinear':
-            num = Add(*[ T**(N-i) * 2**i * c * (z-1)**i * (z+1)**(N-i) for c, i in zip(np[::-1], range(len(np))) ])
-            den = Add(*[ T**(N-i) * 2**i * c * (z-1)**i * (z+1)**(N-i) for c, i in zip(dp[::-1], range(len(dp))) ])
-        if method == 'backward_diff':
-            num = Add(*[ T**(N-i) * c * (z-1)**i * (z)**(N-i) for c, i in zip(np[::-1], range(len(np))) ])
-            den = Add(*[ T**(N-i) * c * (z-1)**i * (z)**(N-i) for c, i in zip(dp[::-1], range(len(dp))) ])
-        if method == 'forward_diff':
-            num = Add(*[ T**(N-i) * c * (z-1)**i for c, i in zip(np[::-1], range(len(np))) ])
-            den = Add(*[ T**(N-i) * c * (z-1)**i for c, i in zip(dp[::-1], range(len(dp))) ])
+def forward_diff(tf, sample_per):
+    r"""
+    Returns falling coefficients of H(z) from numerator and denominator.
+    Where H(z) is the corresponding discretized transfer function,
+    discretized with the forward difference transform method.
+    H(z) is obtained from the continuous transfer function H(s)
+    by substituting s(z) = (z-1)/T into H(s), where T is the
+    sample period.
+    Coefficients are falling, i.e. H(z) = (az+b)/(cz+d) is returned
+    as [a, b], [c, d].
+    Examples
+    ========
+    >>> from sympy.physics.control.lti import TransferFunction, bilinear
+    >>> from sympy.abc import s, L, R, T
+    >>> tf = TransferFunction(1, s*L + R, s)
+    >>> numZ, denZ = forward_diff(tf, T)
+    >>> numZ
+    [T, T]
+    >>> denZ
+    [2*L + R*T, -2*L + R*T]
+    """
+    return gbt(tf, sample_per, 0)
 
-        num_coefs = num.as_poly(z).all_coeffs()
-        den_coefs = den.as_poly(z).all_coeffs()
-
-        return num_coefs, den_coefs
+def backward_diff(tf, sample_per):
+    r"""
+    Returns falling coefficients of H(z) from numerator and denominator.
+    Where H(z) is the corresponding discretized transfer function,
+    discretized with the backward difference transform method.
+    H(z) is obtained from the continuous transfer function H(s)
+    by substituting s(z) =  (z-1)/(T*z) into H(s), where T is the
+    sample period.
+    Coefficients are falling, i.e. H(z) = (az+b)/(cz+d) is returned
+    as [a, b], [c, d].
+    Examples
+    ========
+    >>> from sympy.physics.control.lti import TransferFunction, backward_diff
+    >>> from sympy.abc import s, L, R, T
+    >>> tf = TransferFunction(1, s*L + R, s)
+    >>> numZ, denZ = backward_diff(tf, T)
+    >>> numZ
+    [T, 0]
+    >>> denZ
+    [L + R*T, -L]
+    """
+    return gbt(tf, sample_per, 1)
 
 
 class LinearTimeInvariant(Basic, EvalfMixin):

--- a/sympy/physics/control/tests/test_lti.py
+++ b/sympy/physics/control/tests/test_lti.py
@@ -1274,7 +1274,7 @@ def test_TransferFunction_bilinear():
 def test_TransferFunction_forward_diff():
     # simple transfer function, e.g. ohms law
     tf = TransferFunction(1, a*s+b, s)
-    numZ, denZ = backward_diff(tf, T)
+    numZ, denZ = forward_diff(tf, T)
     # discretized transfer function with coefs from tf.forward_diff()
     tf_test_forward = TransferFunction(s*numZ[0], s*denZ[0]+denZ[1], s)
     # corresponding tf with manually calculated coefs

--- a/sympy/physics/control/tests/test_lti.py
+++ b/sympy/physics/control/tests/test_lti.py
@@ -15,7 +15,7 @@ from sympy.core.containers import Tuple
 from sympy.matrices import ImmutableMatrix, Matrix
 from sympy.physics.control import (TransferFunction, Series, Parallel,
     Feedback, TransferFunctionMatrix, MIMOSeries, MIMOParallel, MIMOFeedback,
-    sample)
+    gbt, bilinear, forward_diff, backward_diff)
 from sympy.testing.pytest import raises
 
 a, x, b, s, g, d, p, k, a0, a1, a2, b0, b1, b2, tau, zeta, wn, T = symbols('a, x, b, s, g, d, p, k,\
@@ -1222,10 +1222,10 @@ def test_TransferFunctionMatrix_functions():
     assert H_5.expand() == \
         TransferFunctionMatrix(((TransferFunction(s**5 + s**3 + s, -s**2 + s, s), TransferFunction(s**2 + 2*s - 3, s**2 + 4*s - 5, s)),))
 
-def test_TransferFunction_sample():
+def test_TransferFunction_gbt():
     # simple transfer function, e.g. ohms law
     tf = TransferFunction(1, a*s+b, s)
-    numZ, denZ = sample(tf, T, 'bilinear')
+    numZ, denZ = gbt(tf, T, 0.5)
     # discretized transfer function with coefs from tf.bilinear()
     tf_test_bilinear = TransferFunction(s*numZ[0]+numZ[1], s*denZ[0]+denZ[1], s)
     # corresponding tf with manually calculated coefs
@@ -1234,7 +1234,16 @@ def test_TransferFunction_sample():
     assert S.Zero == (tf_test_bilinear-tf_test_manual).simplify().num
 
     tf = TransferFunction(1, a*s+b, s)
-    numZ, denZ = sample(tf, T, 'backward_diff')
+    numZ, denZ = gbt(tf, T, 0)
+    # discretized transfer function with coefs from tf.bilinear()
+    tf_test_forward = TransferFunction(s*numZ[0], s*denZ[0]+denZ[1], s)
+    # corresponding tf with manually calculated coefs
+    tf_test_manual = TransferFunction(s*T, s*a-a+b*T, s)
+
+    assert S.Zero == (tf_test_forward-tf_test_manual).simplify().num
+
+    tf = TransferFunction(1, a*s+b, s)
+    numZ, denZ = gbt(tf, T, 1)
     # discretized transfer function with coefs from tf.bilinear()
     tf_test_backward = TransferFunction(s*numZ[0], s*denZ[0]+denZ[1], s)
     # corresponding tf with manually calculated coefs
@@ -1243,10 +1252,43 @@ def test_TransferFunction_sample():
     assert S.Zero == (tf_test_backward-tf_test_manual).simplify().num
 
     tf = TransferFunction(1, a*s+b, s)
-    numZ, denZ = sample(tf, T, 'forward_diff')
+    numZ, denZ = gbt(tf, T, 0.2)
     # discretized transfer function with coefs from tf.bilinear()
+    tf_test_gbt = TransferFunction(s*numZ[0], s*denZ[0]+denZ[1], s)
+    # corresponding tf with manually calculated coefs
+    tf_test_manual = TransferFunction(s*T*0.2+0.8*T, s*(0.2*T*b+1.0*a)+(0.8*T*b-1.0*a), s)
+
+    assert S.Zero == (tf_test_gbt-tf_test_manual).simplify().num
+
+def test_TransferFunction_bilinear():
+    # simple transfer function, e.g. ohms law
+    tf = TransferFunction(1, a*s+b, s)
+    numZ, denZ = bilinear(tf, T)
+    # discretized transfer function with coefs from tf.bilinear()
+    tf_test_bilinear = TransferFunction(s*numZ[0]+numZ[1], s*denZ[0]+denZ[1], s)
+    # corresponding tf with manually calculated coefs
+    tf_test_manual = TransferFunction(s*T+T, s*(T*b+2*a)+T*b-2*a, s)
+
+    assert S.Zero == (tf_test_bilinear-tf_test_manual).simplify().num
+
+def test_TransferFunction_forward_diff():
+    # simple transfer function, e.g. ohms law
+    tf = TransferFunction(1, a*s+b, s)
+    numZ, denZ = backward_diff(tf, T)
+    # discretized transfer function with coefs from tf.forward_diff()
     tf_test_forward = TransferFunction(s*numZ[0], s*denZ[0]+denZ[1], s)
     # corresponding tf with manually calculated coefs
     tf_test_manual = TransferFunction(s*T, s*a-a+b*T, s)
 
     assert S.Zero == (tf_test_forward-tf_test_manual).simplify().num
+
+def test_TransferFunction_backward_diff():
+    # simple transfer function, e.g. ohms law
+    tf = TransferFunction(1, a*s+b, s)
+    numZ, denZ = backward_diff(tf, T)
+    # discretized transfer function with coefs from tf.backward_diff()
+    tf_test_backward = TransferFunction(s*numZ[0]+numZ[1], s*denZ[0]+denZ[1], s)
+    # corresponding tf with manually calculated coefs
+    tf_test_manual = TransferFunction(s*T, s*(T*b+a)-a, s)
+
+    assert S.Zero == (tf_test_backward-tf_test_manual).simplify().num

